annif/backend/nn_ensemble.py

```diff
@@ -97,6 +97,14 @@ class NNEnsembleBackend(
     # defaults for uninitialized instances
     _model = None
 
+    @property
+    def is_trained(self):
+        return super(ensemble.EnsembleBackend, self).is_trained
```

Comment:
Is there any reason to redefine this property here instead of just inheriting it from EnsembleBackend? Or is the problem that it is being inherited from AnnifBackend (via AnnifLearningBackend) instead? Then maybe the order of superclass declarations could be changed?

Reply:
The (re)definition is needed because the property that would be inherited from EnsembleBackend is not applicable here as it does not check the local state (files present in datadir), and I think the NN Ensemble (and PAV) should be considered trained only if they have the model files in datadir. Same goes for the modification time.

The intention here is to use the (default) property from AnnifBackend. Instead of the `super()` call the default implementations could be simply duplicated in here, at least then the lgmt bot would not complain.

Reply:
If the `modification_time` in AnnifBackend wasn't a property but regular method, this could be ofcourse used:
```python
def is_trained(self):
    return backend.AnnifBackend.is_trained(self)
```
As in shown [here](https://stackoverflow.com/questions/18117974/calling-a-parents-parents-method-which-has-been-overridden-by-the-parent) but with property this doesn't work. :/

Reply:
Actually for accessing the property from AnnifBackend is possible with `__get__()`, like:
```python
@property
def is_trained(self):
    return backend.AnnifBackend.is_trained.__get__(self)
```